server/v2/stf/core_services.go

```diff
@@ -64,3 +70,83 @@ func (g gasService) WithBlockGasMeter(ctx context.Context, meter gas.Meter) cont
 	// TODO implement me
 	panic("implement me")
 }
+
+func NewEventService() event.Service {
+	return eventService{}
+}
+
+type eventService struct{}
+
+// EventManager implements event.Service.
+func (eventService) EventManager(ctx context.Context) event.Manager {
+	return &eventManager{ctx.(*executionContext)}
+}
+
+var _ event.Manager = (*eventManager)(nil)
+
+type eventManager struct {
+	executionContext *executionContext
+}
+
+// Emit emits an typed event that is defined in the protobuf file.
+// In the future these events will be added to consensus.
+func (em *eventManager) Emit(tev protoiface.MessageV1) error {
+	res, err := TypedEventToEvent(tev)
+	if err != nil {
+		return err
+	}
+
+	em.executionContext.events = append(em.executionContext.events, res)
+	return nil
+}
+
+// EmitKV emits a key value pair event.
+func (em *eventManager) EmitKV(eventType string, attrs ...event.Attribute) error {
+	attributes := make([]event.Attribute, 0, len(attrs))
+
+	for _, attr := range attrs {
+		attributes = append(attributes, event.NewAttribute(attr.Key, attr.Value))
+	}
```

Comment:
why append when we're given attrs directly?

Reply:
Good point, I shamelessly copied it from the sdk event manager but there it converts to sdk.Attributes.